Sources/Generator/UIGenerator.swift

```diff
@@ -105,7 +129,65 @@ public class UIGenerator: Generator {
                     }
                     try root.children.forEach { try generate(element: $0, superName: "target", containedIn: root) }
                     tempCounter = 1
-                    root.children.forEach { generateConstraints(element: $0, superName: "target") }
+                    root.children.forEach { generateConstraints(element: $0, superName: "target", forUpdate: false) }
+                    if configuration.isLiveEnabled {
+                        l("#endif")
+                    }
+                }
+                l()
+                l("func updateReactantUI()") {
+                    tempCounter = 1
+                    func generateChildVariables(children: [UIElement], guardNames: inout [String]) {
+                        for child in children {
+                            defer {
+                                if let container = child as? UIContainer {
+                                    generateChildVariables(children: container.children, guardNames: &guardNames)
+                                }
+                            }
+
+                            let name: String
+                            if child.field != nil {
+                                continue
+                            } else if let layoutId = child.layout.id {
+                                name = "named_\(layoutId)"
+                            } else {
+                                name = "temp_\(type(of: child))_\(tempCounter)"
+                                tempCounter += 1
+                            }
+
+                            guardNames.append(name)
+                        }
+                    }
+                    var guardNames = ["target"] as [String]
+                    generateChildVariables(children: root.children, guardNames: &guardNames)
+                    if !guardNames.isEmpty {
+                        l("guard ")
+                        for guardName in guardNames {
+                            l("  let \(guardName) = self.\(guardName)\(guardName == guardNames.last ? "" : ",")")
+                        }
+                        l("else { /* FIXME Should we fatalError here? */ return }")
+                    }
+
+                    if configuration.isLiveEnabled {
+                        if configuration.swiftVersion >= .swift4_1 {
+                            l("#if targetEnvironment(simulator)")
+                        } else {
+                            l("#if (arch(i386) || arch(x86_64)) && (os(iOS) || os(tvOS))")
+                        }
+                        // This will register `self` to remove `deinit` from ViewBase
+                        //l("ReactantLiveUIManager.shared.update(target)")
+                        l("#else")
+                    }
+
+                    // TODO: Add conditional properties?
```

Comment:
What did these dead lines do?

Reply:
I'll be honest, it's quite a lot of time since we changed this file with @TadeasKriz and it's mainly his work in this file.

Reply:
No problem, as long as everything still works. I understand that this part of the project is a mess.